src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -399,4 +424,213 @@ namespace AppInstaller::CLI::Workflow
+    {
+        // Ensure that installer type might actually write to ARP, otherwise this is a waste of time
+        auto installer = context.Get<Execution::Data::Installer>();
+
+        if (installer && MightWriteToARP(installer->InstallerType))
+        {
+            std::shared_ptr<ISource> arpSource = context.Reporter.ExecuteWithProgress(
+                [](IProgressCallback& progress)
+                {
+                    return Repository::OpenPredefinedSource(PredefinedSource::ARP, progress);
+                }, true);
+
+            std::vector<std::tuple<Utility::LocIndString, Utility::LocIndString, Utility::LocIndString>> entries;
+
+            for (const auto& entry : arpSource->Search({}).Matches)
+            {
+                auto installed = entry.Package->GetInstalledVersion();
+                entries.emplace_back(std::make_tuple(
+                    entry.Package->GetProperty(PackageProperty::Id),
+                    installed->GetProperty(PackageVersionProperty::Version),
+                    installed->GetProperty(PackageVersionProperty::Channel)));
+            }
+
+            std::sort(entries.begin(), entries.end());
+
+            context.Add<Execution::Data::ARPSnapshot>(std::move(entries));
+        }
+    }
+    CATCH_LOG()
+
+    void ReportARPChanges(Execution::Context& context) try
+    {
+        if (context.Contains(Execution::Data::ARPSnapshot))
+        {
+            const auto& entries = context.Get<Execution::Data::ARPSnapshot>();
+
+            // Open it again to get the (potentially) changed ARP entries
+            std::shared_ptr<ISource> arpSource = context.Reporter.ExecuteWithProgress(
+                [](IProgressCallback& progress)
+                {
+                    return Repository::OpenPredefinedSource(PredefinedSource::ARP, progress);
+                }, true);
+
+            std::vector<ResultMatch> changes;
+
+            for (auto& entry : arpSource->Search({}).Matches)
+            {
+                auto installed = entry.Package->GetInstalledVersion();
+                auto entryKey = std::make_tuple(
+                    entry.Package->GetProperty(PackageProperty::Id),
+                    installed->GetProperty(PackageVersionProperty::Version),
+                    installed->GetProperty(PackageVersionProperty::Channel));
+
+                auto itr = std::lower_bound(entries.begin(), entries.end(), entryKey);
+                if (itr == entries.end() || *itr != entryKey)
+                {
+                    changes.emplace_back(std::move(entry));
+                }
+            }
+
+            // Also attempt to find the entry based on the manifest data
+            const auto& manifest = context.Get<Execution::Data::Manifest>();
+
+            SearchRequest nameAndPublisherRequest;
+
+            // The default localization must contain the name or we cannot do this lookup
+            if (manifest.DefaultLocalization.Contains(Localization::PackageName))

```

Comment:
>manifest.DefaultLocalization.Contains(Localization::PackageName) [](start = 16, length = 64)

PackageName and Publisher is required in default locale in schema

Reply:
But the client code does not assume that manifests are 100% schema compliant.